src/runtime-rs/crates/resource/src/network/network_model/tc_filter_model.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::process::Command;
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+
+use super::{NetworkModel, NetworkModelType};
+use crate::network::NetworkPair;
+
+#[derive(Debug)]
+pub(crate) struct TcFilterModel {}
+
+impl TcFilterModel {
+    pub fn new() -> Result<Self> {
+        Ok(Self {})
+    }
+}
+
+#[async_trait]
+impl NetworkModel for TcFilterModel {
+    fn model_type(&self) -> NetworkModelType {
+        NetworkModelType::TcFilter
+    }
+
+    async fn add(&self, pair: &NetworkPair) -> Result<()> {
+        let tap_name = &pair.tap.tap_iface.name;
+        let virt_name = &pair.virt_iface.name;
+
+        add_qdisc_ingress(tap_name).context("add qdisc ingress for tap link")?;
+        add_qdisc_ingress(virt_name).context("add qdisc ingress")?;
+
+        add_redirect_tcfilter(tap_name, virt_name).context("add tc filter for tap")?;
+        add_redirect_tcfilter(virt_name, tap_name).context("add tc filter")?;
+        Ok(())
+    }
+
+    async fn del(&self, pair: &NetworkPair) -> Result<()> {
+        del_qdisc(&pair.virt_iface.name).context("del qdisc")?;
+        Ok(())
+    }
+}
+
+// TODO: use netlink replace tc command
+fn add_qdisc_ingress(dev: &str) -> Result<()> {
```

Comment:
I think we should make this method async and use tokio command inside

Reply:
Thanks for your review, change to the async.